src/components/app/dialog/dialog.tsx

```diff
@@ -0,0 +1,17 @@
+import {ReactNode, useEffect, useRef} from 'react';
+
+export type DialogProps = ChildProps & { isOpen: boolean, children: ReactNode};
+
+export function Dialog({ className, isOpen, children}: DialogProps): ReactNode {
+    const dialogRef = useRef<HTMLDialogElement>(null);
+    useEffect(() => {
+        if(isOpen) {
+            dialogRef.current?.showModal();
+        } else {
+            dialogRef.current?.close(); }
+        }, [isOpen]);
+
+    return <dialog ref={dialogRef}
```

Comment:
@codelicous Is there a reason why the component filename is lowercase? Because it's native?
In any case, if we're sticking to "daisyui", we can use 
https://daisyui.com/components/modal/